### PR TITLE
fix: server port and host

### PR DESCRIPTION
**Pull request by ycjcl868**
支持 `devServer` 配置 `port`、`host`

##### Checklist

- [x] `npm test` passes
- [ ] tests are included
- [ ] documentation is changed or added
- [ ] commit message follows commit guidelines

##### Description of change

- any feature?
- close https://github.com/umijs/umi/ISSUE_URL

### CODE REVIEW

**Comment by sorrycc** [comment]
ci 是不是不稳定？

**Comment by ycjcl868** [comment]
server 的测试依赖端口，我调整下

File: packages/preset-built-in/src/plugins/commands/dev/mock/createMiddleware.test.ts [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import { join } from 'path';
 import { writeFileSync } from 'fs';
 import { Service } from '@umijs/core';
+import portfinder from 'portfinder';
```
Comment (sorrycc) [inline packages/preset-built-in/src/plugins/commands/dev/mock/createMiddleware.test.ts]: portfinder 在 `@umijs/utils` 里有了。

Comment (ycjcl868) [reply packages/preset-built-in/src/plugins/commands/dev/mock/createMiddleware.test.ts]: test case 里单独起了一个 proxy server

Comment (ycjcl868) [reply packages/preset-built-in/src/plugins/commands/dev/mock/createMiddleware.test.ts]: 我从 `@umijs/utils` 志

**Pull request merged**